src/pages/trader-tools/swap-calculator/_swap-calculator.tsx

```diff
@@ -56,6 +56,131 @@ import { Flex, Show } from 'components/containers'
 import Input from 'components/form/input'
 import RightArrow from 'images/svg/tools/black-right-arrow.svg'
 
+type FormikErrors<Values> = {
+    [K in keyof Values]?: Values[K] extends string[]
+        ? Values[K][number] extends Record<string, unknown>
+            ? FormikErrors<Values[K][number]>[] | string | string[]
+            : string | string[]
+        : Values[K] extends Record<string, unknown>
+        ? FormikErrors<Values[K]>
+        : string
+}
+
+type FormikTouched<Values> = {
+    [K in keyof Values]?: Values[K] extends string[]
+        ? Values[K][number] extends Record<string, unknown>
+            ? FormikTouched<Values[K][number]>[]
+            : boolean
+        : Values[K] extends Record<string, unknown>
+        ? FormikTouched<Values[K]>
+        : boolean
+}
```

Comment:
Since these were used in multiple places, let's move them into a separate type file to be shared across the calculators. Good job on these types 👍